lib/datadog/search-syntax/src/grammar.rs

```diff
@@ -44,7 +44,7 @@ impl QueryVisitor {
                             // If our conjunction is AND and the previous clause was
                             // just a SHOULD, we make the previous clause a MUST and
                             // our new clause will also be a MUST
-                            let mut lastitem = clauses.last_mut().unwrap();
+                            let lastitem = clauses.last_mut().unwrap();
```

Comment:
does this break in THIS version of rust?  seems like an odd incompatability

Reply:
This is a warning even in the most recent version of rustc, but at some point the vector upstream decided to ban warnings, like `-Werror` in GCC:
```
2023-11-20 16:25:05,353 - atomic_reactor.tasks.binary_container_build - INFO - error: variable does not need to be mutable
2023-11-20 16:25:05,353 - atomic_reactor.tasks.binary_container_build - INFO -   --> lib/datadog/search-syntax/src/grammar.rs:47:33
2023-11-20 16:25:05,353 - atomic_reactor.tasks.binary_container_build - INFO -    |
2023-11-20 16:25:05,353 - atomic_reactor.tasks.binary_container_build - INFO - 47 | ...                   let mut lastitem = clauses.last_mut().unwrap();
2023-11-20 16:25:05,353 - atomic_reactor.tasks.binary_container_build - INFO -    |                           ----^^^^^^^^
2023-11-20 16:25:05,353 - atomic_reactor.tasks.binary_container_build - INFO -    |                           |
2023-11-20 16:25:05,353 - atomic_reactor.tasks.binary_container_build - INFO -    |                           help: remove this `mut`
2023-11-20 16:25:05,353 - atomic_reactor.tasks.binary_container_build - INFO -    |
2023-11-20 16:25:05,353 - atomic_reactor.tasks.binary_container_build - INFO - note: the lint level is defined here
2023-11-20 16:25:05,353 - atomic_reactor.tasks.binary_container_build - INFO -   --> lib/datadog/search-syntax/src/lib.rs:1:9
2023-11-20 16:25:05,353 - atomic_reactor.tasks.binary_container_build - INFO -    |
2023-11-20 16:25:05,353 - atomic_reactor.tasks.binary_container_build - INFO - 1  | #![deny(warnings)]
2023-11-20 16:25:05,353 - atomic_reactor.tasks.binary_container_build - INFO -    |         ^^^^^^^^
2023-11-20 16:25:05,353 - atomic_reactor.tasks.binary_container_build - INFO -    = note: `#[deny(unused_mut)]` implied by `#[deny(warnings)]`
```